src/graph-layouts.jl

```diff
@@ -0,0 +1,89 @@
+export distgraph, distdigraph, num_vertices, vertices, adj
+###
+# Graph data structure definitions.
+###
+
+# Allowed Graph datastructures:
+typealias AdjacencyList Array{Array{Int, 1}, 1}
+typealias AdjacencyMatrix AbstractArray{Int, 2}                      # Covers sparse matrices
+typealias GraphStruct Union{AdjacencyList, AdjacencyMatrix}
+
+
+# Generic (undirected or directed, up to the user to decide) Graph
+type DistGraph{S<:GraphStruct}
+    nv::Int                            # not really required, but kept for convenience
+    vertices::UnitRange{Int}
```

Comment:
So far I don't see why we need this field.

Am thinking of allowing `slice` to return some metadata about each chunk and then storing that metadata in the DistMemory. So that each layout type can put some metadata along with the chunks to facilitate things like `length`, `getindex